### PR TITLE
add p6spy driver support for proxy backend

### LINKED ISSUE #6266: jdbc url not support p6spy driver
## Bug Report

jdbc url not support p6spy driver  examples like: jdbc:p6spy:mysql://credigo.ct2ud7p

**exceptions:** 
org.apache.shardingsphere.infra.exception.ShardingSphereException: Cannot resolve JDBC url `jdbc:p6spy:mysql:xxx`. Please implements `org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.spi.JDBCDriverURLRecognizer` and add to SPI.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?
5.x 
### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy
### Expected behavior
  mysql 

### Actual behavior
org.apache.shardingsphere.infra.exception.ShardingSphereException: Cannot resolve JDBC url `jdbc:p6spy:mysql:xxx`. Please implements `org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.spi.JDBCDriverURLRecognizer` and add to SPI.

### Reason analyze (If you can)
**current implement depends on JDBCDriverURLRecognizer , it just use url  prefix match to check which driver suitable, 
can't support such as  jdbc:p6spy:mysql:xxx**

### Example codes for reproduce this issue (such as a github link).

JDBCDriverURLRecognizerEngine.getJDBCDriverURLRecognizer("jdbc:p6spy:mysql:xxx");

### PR DESCRIPTION
**Pull request by xbkaishui**
Fixes #6266.

Changes proposed in this pull request:
- support p6spy driver, p6spy is a wrapper of database driver

### CODE REVIEW

**Comment by tristaZero** [comment]
Hi @xbkaishui Glad to see you again. I believe @tuohai666 will give it a review carefully, so please be patient.

Trista

**Comment by xbkaishui** [comment]
Yes, no worry, I am very appreciate @tuohai666 can review my code,  I  learned so much from the previous PR comments.

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/JDBCDriverURLRecognizerEngine.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
 import lombok.AccessLevel;
 import lombok.NoArgsConstructor;
 import org.apache.shardingsphere.infra.exception.ShardingSphereException;
+import org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.spi.JDBCDriverComposeURLRecognizer;
```
Comment (tuohai666) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/JDBCDriverURLRecognizerEngine.java:23]: Please update all config-*.yaml

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/resources/META-INF/services/org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.spi.JDBCDriverURLRecognizer [hunk 2]
```diff
@@ -20,3 +20,4 @@ org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.Postg
 org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.OracleRecognizer
 org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.SQLServerRecognizer
 org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.H2Recognizer
+org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.P6SpyDriverRecognizer
```
Comment (tuohai666) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/resources/META-INF/services/org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.spi.JDBCDriverURLRecognizer:23]: Need an empty line.

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/P6SpyDriverRecognizerTest.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public class P6SpyDriverRecognizerTest {
+    
+    private final P6SpyDriverRecognizer recognizer = new P6SpyDriverRecognizer();
+    
+    @Test
+    public void getURLPrefixes() throws Exception {
```
Comment (tuohai666) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/P6SpyDriverRecognizerTest.java:32]: Redundant exception.

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/P6SpyDriverRecognizerTest.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        assertThat(recognizer.getURLPrefixes(), is(Collections.singletonList("jdbc:p6spy:")));
+    }
+    
+    @Test
+    public void getDriverClassName() throws Exception {
```
Comment (tuohai666) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/P6SpyDriverRecognizerTest.java:37]: Redundant exception.

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/PostgreSQLRecognizer.java [hunk 5]
```diff
@@ -27,9 +28,11 @@
  */
 public final class PostgreSQLRecognizer implements JDBCDriverURLRecognizer {
     
+    private PostgreSQLDatabaseType databaseType = new PostgreSQLDatabaseType();
```
Comment (tuohai666) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/PostgreSQLRecognizer.java:31]: Field 'databaseType' may be 'final'

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/SQLServerRecognizer.java [hunk 6]
```diff
@@ -27,9 +28,11 @@
  */
 public final class SQLServerRecognizer implements JDBCDriverURLRecognizer {
     
+    private SQLServerDatabaseType databaseType = new SQLServerDatabaseType();
```
Comment (tuohai666) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/SQLServerRecognizer.java:31]: Field 'databaseType' may be 'final'

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/H2Recognizer.java [hunk 7]
```diff
@@ -27,9 +28,11 @@
  */
 public final class H2Recognizer implements JDBCDriverURLRecognizer {
     
+    private H2DatabaseType databaseType = new H2DatabaseType();
```
Comment (tuohai666) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/H2Recognizer.java:31]: Field 'databaseType' may be 'final'

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/MySQLRecognizer.java [hunk 8]
```diff
@@ -27,9 +28,11 @@
  */
 public final class MySQLRecognizer implements JDBCDriverURLRecognizer {
     
+    private MySQLDatabaseType databaseType = new MySQLDatabaseType();
```
Comment (tuohai666) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/MySQLRecognizer.java:31]: Field 'databaseType' may be 'final'

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/OracleRecognizer.java [hunk 9]
```diff
@@ -27,9 +28,11 @@
  */
 public final class OracleRecognizer implements JDBCDriverURLRecognizer {
     
+    private OracleDatabaseType databaseType = new OracleDatabaseType();
```
Comment (tuohai666) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/OracleRecognizer.java:31]: Field 'databaseType' may be 'final'

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/JDBCDriverURLRecognizerEngine.java (hunk 1 above)
Comment (tuohai666) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/JDBCDriverURLRecognizerEngine.java:23]: add spy.properties

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/PostgreSQLRecognizer.java (hunk 5 above)
Comment (xbkaishui) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/PostgreSQLRecognizer.java:31]: done

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/H2Recognizer.java [hunk 10]
```diff
@@ -27,9 +28,11 @@
  */
 public final class H2Recognizer implements JDBCDriverURLRecognizer {
     
+    private final H2DatabaseType databaseType = new H2DatabaseType();
+    
     @Override
     public String getDatabaseType() {
```
Comment (tuohai666) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/H2Recognizer.java:34]: Can this method revert to origin?

File: shardingsphere-proxy/shardingsphere-proxy-bootstrap/src/main/resources/spy.properties [hunk 11]
```diff
@@ -0,0 +1,247 @@
+#################################################################
```
Comment (tuohai666) [inline shardingsphere-proxy/shardingsphere-proxy-bootstrap/src/main/resources/spy.properties:1]: Consider delete this file, let user add this is better.

Comment (xbkaishui) [reply shardingsphere-proxy/shardingsphere-proxy-bootstrap/src/main/resources/spy.properties:1]: fine

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/H2Recognizer.java (hunk 10 above)
Comment (xbkaishui) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/recognizer/impl/H2Recognizer.java:34]: I saw the code contains so many literals of databaseType,  is it better to refers only *DatabaseType class ?

**Pull request merged**